ably-ios/ARTRealtimeChannel.m

```diff
@@ -371,8 +371,10 @@ - (void)onMessage:(ARTProtocolMessage *)message {
             NSError *error = nil;
             msg = [msg decodeWithEncoder:dataEncoder error:&error];
             if (error != nil) {
-                [self.logger error:@"ARTRealtimeChannel: error decoding data: %@", error];
-                [NSException raise:NSInvalidArgumentException format:@"ARTRealtimeChannel: error decoding data: %@", error];
+                ARTErrorInfo *errorInfo = [ARTErrorInfo wrap:(ARTErrorInfo *)error.userInfo[NSLocalizedFailureReasonErrorKey] prepend:@"Failed to decode data: "];
+                [self.logger error:@"%@", errorInfo.message];
+                _errorReason = errorInfo;
+                [self emit:self.state with:errorInfo];
```

Comment:
Does this mean on every error we are emitting the the current state again?  If you look at http://docs.ably.io/client-lib-development-guide/features/#RTL2c, it states we emit an error, but not emit the same state. It's a bit of a fudge because error is not a state, yet we emit an error from the eventemitter, see https://github.com/ably/ably-ruby/blob/master/spec/acceptance/realtime/channel_spec.rb#L174-L178 for an example. I am not sure how we have an equivalent for iOS, perhaps one to discuss?